### PR TITLE
logging: unify pkg/log and plugin/log

### PR DESCRIPTION
**Pull request by miekg**
Default to using pkg/log for all logging output and make that always use
UTC.

Signed-off-by: Miek Gieben <miek@miek.nl>

### CODE REVIEW

File: plugin/log/README.md [hunk 1]
```diff
@@ -7,7 +7,7 @@
 ## Description
 
 By just using *log* you dump all queries (and parts for the reply) on standard output. Options exist
-to tweak the output a little.
+to tweak the output a little. The timezone is used is always UTC.
```
Comment (fturib) [inline plugin/log/README.md]: typo. "The timezone is always UTC"

File: coremain/run.go [hunk 2]
```diff
@@ -62,7 +62,7 @@ func Run() {
 	}
 
 	log.SetOutput(os.Stdout)
-	log.SetFlags(log.LstdFlags)
+	log.SetFlags(log.Ldate | log.Ltime | log.LUTC)
```
Comment (fturib) [inline coremain/run.go]: Changing to UTC will surprise everyone, and the date does not say the TZ.
So maybe, while waiting we implement a complete date with TZ in our pkg/log, we can output an info here:
```
	clog.Info("log output - all dates are UTC")
```

Result is:

```
API server listening at: 127.0.0.1:55958
2018/10/29 14:51:55 [INFO] log output - all dates are UTC
.:1053
.:1054
2018/10/29 14:51:55 [INFO] CoreDNS-1.2.5
2018/10/29 14:51:55 [INFO] darwin/amd64, go1.10.2, 
CoreDNS-1.2.5
darwin/amd64, go1.10.2, 
2018/10/29 14:51:55 [INFO] plugin/reload: Running configuration MD5 = 67fbea4124263c10c9c1aa0fb55a538f
```

**Review by fturib — COMMENTED** [review]
See my comments about UTC information proposition.
+ glitch in one comment.

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [coredns/coredns] logging: unif..." ]
>+to tweak the output a little. The timezone is used is always UTC.
>
>typo. "The timezone is always UTC"

done.

>> @@ -62,7 +62,7 @@ func Run() {
> 	}
>
> 	log.SetOutput(os.Stdout)
>-	log.SetFlags(log.LstdFlags)
>+	log.SetFlags(log.Ldate | log.Ltime | log.LUTC)
>
>Changing to UTC will surprise everyone, and the date does not say the TZ.

it does not say the timezone because it's UTC, but I agree the std log package 
is pretty brain dead in this regard. Logging in local time with a time zone is 
even crazier.

~~~
>2018/10/29 14:51:55 [INFO] log output - all dates are UTC
~~~

this doesn't add much to saying it in the README (i.e. both will not be read)

/Miek

--
Miek Gieben

**Comment by miekg** [comment]
I'll update this to spit the date in w/ a timezone and RFC3339

**Review by fturib — APPROVED** [review]
LGTM

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [coredns/coredns] logging: unif..." ]
>fturib approved this pull request.
>
>LGTM

thanks we have to decide how backwards incompatible this is; can we just push it 
in the next or will this lead to any degraded experience.

**Comment by fturib** [comment]
I was thinking that if you enable Log plugin, then you already know you will have no perf ...
But maybe I am wrong.

I know Log is not enable on the default Corefile for Kubernetes  

@rdrozhdzh, @drazhanski, @huynhmb359  - will you be impacted if the log is less performant  ?

File: plugin/pkg/log/log.go [hunk 3]
```diff
@@ -1,31 +1,36 @@
...
 var D bool
 
+// RFC3339Milli doesn't exist, invent it here.
+func clock() string { return time.Now().Format("2006-01-02T15:04:05.999Z07:00") }
+
 // logf calls log.Printf prefixed with level.
 func logf(level, format string, v ...interface{}) {
-	s := level + fmt.Sprintf(format, v...)
-	golog.Print(s)
+	golog.Print(clock() + " " + level + fmt.Sprintf(format, v...))
```
Comment (rdrozhdzh) [inline plugin/pkg/log/log.go]: minor optimizations -
1. `" " + level` - the leading space can be added to constants like `info = " [INFO] "`
2. I suppose the code `golog.Print(clock(), level, fmt.Sprintf(format, v...))` will concatenate strings more effectively

the same for `func log()` below

**Review by rdrozhdzh — APPROVED** [review]
LGTM
I don't think we'll get any significant performance penalty with this implementation. Earlier, golang log library did time formatting for us, now we do similar formatting ourselves.

**Review by jameshartig — APPROVED** [review]
LGTM

**Comment by drazhanski** [comment]
> @drazhanski - will you be impacted if the log is less performant ?
I don't expect performance problems with this implementation. And usually we log only error/denial classes  of responses, so it can't be so critical.

**Comment by miekg** [comment]
I'll make those changes and then merge.

File: plugin/pkg/log/log.go (hunk 3 above)
Comment (miekg) [reply plugin/pkg/log/log.go]: done

**Pull request merged**